contracts/PaymentSplitter.sol

```diff
@@ -0,0 +1,241 @@
+ *
+ * The split can be in equal parts or in any other arbitrary proportion. The way this is specified is by assigning each
+ * account to a number of shares. Of all the Ether that this contract receives, each account will then be able to claim
+ * an amount proportional to the percentage of total shares they were assigned.
+ *
+ * `PaymentSplitter` follows a _pull payment_ model. This means that payments are not automatically forwarded to the
+ * accounts but kept in this contract, and the actual transfer is triggered as a separate step by calling the {release}
+ * function.
+ *
+ * NOTE: This contract assumes that ERC20 tokens will behave similarly to native tokens (Ether). Rebasing tokens, and
+ * tokens that apply fees during transfers, are likely to not be supported as expected. If in doubt, we encourage you
+ * to run tests before sending real value to this contract.
+ */
+contract PaymentSplitter is Context {
+    event PayeeAdded(address account, uint256 shares);
+    event PayeeRemoved(address account, uint256 shares);
+    event PaymentReleased(address to, uint256 amount);
+    event ERC20PaymentReleased(
+        IERC20 indexed token,
+        address to,
+        uint256 amount
+    );
+    event PaymentReceived(address from, uint256 amount);
+
+    uint256 private _totalReleased;
+
+    // Revshare amount
+    uint256 internal _totalDepositedAmount;
+    uint256 internal _totalDepositedReleased;
+
+    mapping(address => uint256) public _shares;
+    mapping(address => uint256) private _released;
+    address[] internal _payees;
+
+    mapping(IERC20 => uint256) private _erc20TotalReleased;
+    mapping(IERC20 => mapping(address => uint256)) private _erc20Released;
+
+    /**
+     * @dev Creates an instance of `PaymentSplitter` where each account in `payees` is assigned the number of shares at
+     * the matching position in the `shares` array.
+     *
+     * All addresses in `payees` must be non-zero. Both arrays must have the same non-zero length, and there must be no
+     * duplicates in `payees`.
+     */
+    constructor(address payee_) {
+        _addPayee(payee_, 100);
+    }
+
+    /**
+     * @dev The Ether received will be logged with {PaymentReceived} events. Note that these events are not fully
+     * reliable: it's possible for a contract to receive Ether without triggering this function. This only affects the
+     * reliability of the events, and not the actual splitting of Ether.
+     *
+     * To learn more about this see the Solidity documentation for
+     * https://solidity.readthedocs.io/en/latest/contracts.html#fallback-function[fallback
+     * functions].
+     */
+    receive() external payable virtual {
+        emit PaymentReceived(_msgSender(), msg.value);
+    }
+
+    /**
+     * @dev Getter for the total amount of Ether already released.
+     */
+    function totalReleased() public view returns (uint256) {
```

Comment:
This and below publicly declared functions could be external

Reply:
Not sure I can as I'm calling it internally [here](https://github.com/simpleweb/open-format-contracts/pull/3/files/99cc69377feb12717d3c1a13d20790c99eb02a21#diff-4338a100ba79b4498cb885aebc8f450c955b7255fa0fb9bcef07dc1e27b86184R171)